tests/version_test.py

```diff
@@ -30,6 +30,7 @@
   (?P<version>[0-9]+\.[0-9]+\.[0-9]+) # main version; like '0.4.16'
   (?:\.dev(?P<dev>[0-9]+))?           # optional dev version; like '.dev20230908'
   (?:\+(?P<local>[a-zA-Z0-9_]+))?     # optional local version; like '+g6643af3c3'
+  (?:\.(?P<suffix>[a-zA-Z0-9_]+))?    # optional custom suffix; like 'test'
```

Comment:
Another comment here: this regex is wrong, because it would allow version strings like `0.4.24.test` which is invalid under the PEP440 rules. You'll need to make the `suffix` only appear when the `+` appears – probably easiest would be to just wrap it into the existing `local` version.

Reply:
Thanks. That makes a lot sense. So I will just append the regex string to the local version to make it like a suffix to the local version.

Reply:
To avoid some back-and-forth for this test regex, do you think the following make sense?

`(?:\+(?P<local>[a-zA-Z0-9_]+)\.([a-zA-Z0-9_]+))?`

The case is that I don't want to allow things like `a.b.c.d`.

Reply:
And I guess as this is just for testing and in the tests, we don't actually verify the correctness of everything that conforms to the PEP standard. If you think something simple like `(?:\+(?P<local>[a-zA-Z0-9_\.]+)))?` is OK, we can do that.

Reply:
Sorry this is not correct `(?:\+(?P<local>[a-zA-Z0-9_]+)\.([a-zA-Z0-9_]+))?`. Then just add the dot into the string?

Reply:
https://peps.python.org/pep-0440/

They already have one: (?:\+(?P<local>[a-z0-9]+(?:[-_\.][a-z0-9]+)*))?       # local version

We just use this one instead I guess.

Reply:
![image](https://github.com/user-attachments/assets/e988a3bf-d4ea-4057-9d71-8a1b03945fe0)

I guess all these are allowed with the official regex.

Reply:
I updated to the official version in the PEP 440 page. Hope this resolves your concern.

Reply:
As noted in the comment above the regex, we deliberately do not support the full PEP 440 syntax.

In this case I'd suggest using something like this: `(?:\+(?P<local>[a-zA-Z0-9_.]+))?` (note the addition of `.` compared to the existing expression). All you're doing is augmenting the local version with a dot and some additional characters, there's not need to make it more complicated than that.

Reply:
OK. Fine, updated to what you suggested.

Your comment just says skipping the pre- and post- release but the history of your code doesn't mention how to deal with the local version. I didn't extend the regex used in the test for pre and post versions but just extends to use the official local version regex. Sorry I didn't see any explanation on why it is delibrately not supported.

To be fair, I think this regex is only used in tests but has no control over what users want to put inside the CUSTOM_SUFFIX env. So as long as you told me which regex to use, I am fine with it. All I need is just this feature which is blocking us implementing better AOT cache control.